### PR TITLE
fix(ls): preserve hierarchy for -R + passthrough incompatible flags (#714)

### LINKED ISSUE #714: `rtk ls -R` flattens recursive output and loses hierarchy context
When running `rtk ls -R <directory>`, the output is flattened into a list of names without enough hierarchy context to determine where each entry belongs.

I understand the goal may be to keep the output compact and token-efficient, and I am not asking for full relative paths for every file.

However, the current output removes too much structure. It becomes ambiguous, especially when multiple subdirectories contain files with the same name.

## Steps to reproduce

Given a structure like this:

```text
docs/
docs/group-a/
docs/group-a/overview.md
docs/group-a/details.md
docs/group-b/
docs/group-b/overview.md
docs/group-b/examples.md
docs/group-b/nested/
docs/group-b/nested/notes.md
docs/empty-dir/
docs/readme.md
```

Run:

```bash
rtk ls -R docs/
```

## Actual behavior

The output is flattened and loses directory context:

```text
group-a/
group-b/
nested/
empty-dir/
readme.md  2.4K
overview.md  1.9K
details.md  3.1K
overview.md  2.2K
examples.md  4.0K
notes.md  1.1K
```

From this output, it is not possible to determine:

* which directory each file belongs to
* whether `nested/` is inside `group-b/` or at the root
* whether later entries are still inside a previously listed directory

## Expected behavior

The output should remain compact, but preserve enough hierarchy context to be unambiguous.

Full relative paths are not required. However, the output must provide a way to understand directory structure and file ownership.

## Example acceptable outputs

A minimal tree format would already solve the ambiguity while staying compact:

```text
readme.md  2.4K
group-a/
 overview.md  1.9K
 details.md  3.1K
group-b/
 overview.md  2.2K
 examples.md  4.0K
 nested/
  notes.md  1.1K
empty-dir/
```

Another compact option would be a section-based format:

```text
readme.md  2.4K
group-a/
overview.md  1.9K
details.md  3.1K
group-b/
overview.md  2.2K
examples.md  4.0K
group-b/nested/
notes.md  1.1K
empty-dir/
```

Both formats preserve hierarchy context without requiring full relative paths on every file entry.

The exact format is flexible. The key requirement is that recursive output must make directory context explicit.

## Why this matters

Without hierarchy context, recursive output is ambiguous for both humans and scripts, especially when duplicate file names exist in different directories.

## Environment

* OS: Windows
* Shells tested: PowerShell, Git Bash, WSL
* RTK version: `0.30.1`

## Additional context

This does not appear to be shell-specific. The issue seems to be in how recursive output is generated or rendered, as directory hierarchy information is lost.

### PR DESCRIPTION
**Pull request by pszymkowiak**
Supersedes #799 — that PR was based on the pre-refacto `src/ls.rs` and could not be rebased cleanly after the move to `src/cmds/system/ls.rs` with the new `runner::run_filtered` architecture. Ported the fix to the current codebase.

Closes #714

## Changes

1. **Recursive hierarchy** (#714)
   - New `compact_ls_recursive()` parses `ls -laR` sections and emits:
     - Subsection headers: `/path/to/dir:`
     - Children indented 2 spaces under their header
   - LLM agents can now tell which files live in which directory. Before, everything was collapsed into a flat list.

2. **Incompatible flags passthrough**
   - Flags `-1`, `-d`, `-F`, `-C`, `-m`, `-x`, `-i`, `-p` change ls output format enough that our `ls -la` parser breaks.
   - Now pass those flags through to raw ls so users still get the expected behavior.

## Before/After

```
BEFORE (flat):               AFTER (hierarchy):
empty-dir/                   empty-dir/
group-a/                     group-a/
group-b/                     group-b/
nested/                      nested/
readme.md  7B                readme.md  7B
details.md  8B
overview.md  11B             /path/group-a:
examples.md  9B                details.md  8B
overview.md  11B               overview.md  11B
notes.md  6B
                             /path/group-b:
                               examples.md  9B
                               overview.md  11B

                             /path/nested:
                               notes.md  6B
```

## Tests

- [x] `cargo test` — 1386 passed, 0 failed, 0 regressions
- [x] 2 new tests for `compact_ls_recursive` (hierarchy + empty subdir edge case)
- [x] Manual: `rtk ls -R`, `rtk ls -1`, `rtk ls` all work as expected
- [x] `cargo fmt --all` + `cargo clippy --all-targets` clean

### CODE REVIEW

**Comment by pszymkowiak** [comment]
> [w] **wshm** · Automated triage by AI

## 📊 Automated PR Analysis

| | |
|---|---|
| 🐛 **Type** | `bug-fix` |
| 🟡 **Risk** | `medium` |

### Summary

Fixes recursive ls output to preserve directory hierarchy by introducing compact_ls_recursive() that emits subsection headers with indented children, and adds a passthrough path for flags incompatible with the ls -la parser. This ensures LLM agents can distinguish which files belong to which directory instead of seeing a flat list.

### Review Checklist
- [x] Tests present
- [ ] Breaking change
- [ ] Docs updated

**Linked issues:** #714

---
*Analyzed automatically by [wshm](https://github.com/wshm-dev/wshm)* · This is an automated analysis, not a human review.

File: src/cmds/system/ls.rs [hunk 1]
```diff
@@ -34,6 +34,49 @@ pub fn run(args: &[String], verbose: u8) -> Result<i32> {
...
+        } else {
+            for p in &paths {
+                raw_cmd.arg(p);
+            }
+        }
+        let output = raw_cmd.output()?;
+        print!("{}", String::from_utf8_lossy(&output.stdout));
+        if !output.status.success() {
+            eprint!("{}", String::from_utf8_lossy(&output.stderr));
+        }
```
Comment (KuSh) [inline src/cmds/system/ls.rs:72]: Shouldn't we output stderr if anything was outputted there, even on success?

File: src/cmds/system/ls.rs [hunk 2]
```diff
@@ -232,6 +279,114 @@ fn compact_ls(raw: &str, show_all: bool) -> (String, String) {
     (entries, summary)
 }
 
+/// Parse `ls -laR` output into hierarchical format.
+///
+/// `ls -laR` emits sections separated by blank lines, where each section is
```
Comment (KuSh) [inline src/cmds/system/ls.rs:284]: As we already indent hierarchical content I'm not sure the blank lines are really needed.

**Review by KuSh — COMMENTED** [review]
Otherwise LGTM!
